### ISSUE TITLE
Wak v2.0.0.a Rest api broken (BUG)

### ISSUE THREAD

**Issue by vegarringdal** [Wakanda/wakanda-issues#148]
@hamzahik, @fredGandelot 

Wak v2.0.0.a Rest Api is broken, when running $update with $refresh to trigger update without save/validation
Only issue in the beta.

In my own datasource I bind I get issues on the beta, alias attribute does no get updated
Datasource:https://github.com/vegarringdal/aurelia-wak

# This is what I send:
![image](https://cloud.githubusercontent.com/assets/2901416/22523614/4de808bc-e8c0-11e6-8274-1296489a7bfe.png)

----

# On beta I get back this:
The country is **NOT** updated (related entity) 👎 
![image](https://cloud.githubusercontent.com/assets/2901416/22523844/19d09c32-e8c1-11e6-924b-9b9502d064f3.png)

---

# On Wakanda Studio  1.1.2
I  get correct return 👍 

![image](https://cloud.githubusercontent.com/assets/2901416/22523811/feea66c8-e8c0-11e6-8cbf-1e98f758e08a.png)

# Model:

![image](https://cloud.githubusercontent.com/assets/2901416/22523917/65a6f7b4-e8c1-11e6-89dd-cde90e3647db.png)

**Comment by vegarringdal** [comment]
Looks like when this bug happends all my numbers and long attributes in the class is dropped/ set to null
Really weird

**Comment by fredGandelot** [comment]
Hi @vegarringdal  We reproduced this one toon, but seems not related to a number nor long related issue.
As u can guess. I give high priority to this fix. 
I 'll keep u posted here , so u can follow up how the fix progress
Regard.

**Comment by vegarringdal** [comment]
@fredGandelot 
Great, I created another issue with the numbers/long issues dropping

Any other changes made to the rest api, there is a few places where it could be improved

**Comment by vegarringdal** [comment]
@fredGandelot what the status of this, new beta coming soon ?

**Comment by fredGandelot** [comment]
@vegarringdal  It's fixed and integrated in the `dev branch`

Btw, yes it's coming soon :)

**Comment by vegarringdal** [comment]
See this one is still open, it was not fixed in 2.0 ?

**Comment by fredGandelot** [comment]
@vegarringdal  it's fixed :)

**Comment by vegarringdal** [comment]
@fredGandelot 

Thx

**Comment by vegarringdal** [comment]
@fredGandelot 
Looks like this is not fixed in the beta 2.0.1

When Ive binded and do update rest with refresh the alias attribute is not updated

![image](https://cloud.githubusercontent.com/assets/2901416/24876278/5c3402a4-1e2b-11e7-9eb3-42158d231c7b.png)

**Comment by vegarringdal** [comment]
@fredGandelot @hamzahik 

Complete data on the issue:

When I do bind and the update/refresh data:
Headers:
![image](https://cloud.githubusercontent.com/assets/2901416/24876534/5d1b5f18-1e2c-11e7-88ba-d53b95073a7d.png)

![image](https://cloud.githubusercontent.com/assets/2901416/24876649/c0d0d42a-1e2c-11e7-9b19-055dff34263e.png)

Saving the entity corrects it, but really does not much help when we cant do binding and refresh to get the alias attributes updated

![image](https://cloud.githubusercontent.com/assets/2901416/24876674/dd4715a6-1e2c-11e7-8846-838c13b27f58.png)

**Comment by fredGandelot** [comment]
Hi @vegarringdal 

I'm using 2.0.1 (no beta) and for me it appears to be fix.

```
var entity= {
__KEY:"2",
__STAMP:10,
countryRef:{__KEY:"1",__STAMP:1}
}

var xhr=new XMLHttpRequest();
function cb(){
	if(xhr.readyState===4){
		console.log(xhr.responseText)
	}
}

xhr.onreadystatechange=cb;
xhr.open('POST','http://127.0.0.1:8081/REST/Person/?$method=update',true);
xhr.send(JSON.stringify(entity));
```



And here's the answer :

![capture d ecran 2017-04-11 a 10 04 47](https://cloud.githubusercontent.com/assets/4920957/24898811/6385f710-1e9e-11e7-9a0a-95730bc13773.png)

**Comment by vegarringdal** [comment]
@fredGandelot 
You are doing a update (save onky) and not adding ```$refresh="true"```  to just get updated /refreshed with binding/alias attributes updated without saving. (guess this is the same as validate only)

So its still a bug

**Comment by fredGandelot** [comment]
@vegar my bad , cause the bug we fixed was also occurring when saving the entity (on the answer the alias wasn'(t updated too ..)
The specific fix for refresh will be handle asap.

**Comment by vegarringdal** [comment]
@fredGandelot 
Great:)
can you set this issue as open until its fixed

My modified list of rest api, not all shown in the docs, want my datasource to support all in the end, but will take some time
https://github.com/vegarringdal/aurelia-wak/blob/master/src/wakdb/WakRestUtil.ts

**Comment by fredGandelot** [comment]
Dev done 

@vegarringdal  => Except the fix to be integreated in `wakanda 2.1`

**Comment by vegarringdal** [comment]
Great :-)
